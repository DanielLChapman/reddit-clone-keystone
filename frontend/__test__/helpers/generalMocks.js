import { CURRENT_USER_QUERY } from '../../components/User';
import { fakeTextPost, fakeUser} from '../../lib/testUtils';

const fakeUserObj = fakeUser();
const notSignedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: null } },
    },
  ];
  
const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() /*{
        
            id: fakeUserObj.id,
            username: fakeUserObj.username,
            email: fakeUserObj.email,
            name:fakeUserObj.name,
            posts: [],
            comments: [],
            moderating: [],
            owner:[],
            subreddits:[],
            postvotes:[],
            commentvotes: [],
        
      }*/ } },
    },
  ];
  
const signedInMocksWithPosts = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: {
        data: {
          authenticatedItem: fakeUser({
            posts: [fakeTextPost()],
          }),
        },
      },
    },
];

export {
    notSignedInMocks,
    signedInMocks,
    signedInMocksWithPosts
}

/* 

{
        id
        username
        email
        name
        posts {
          votes {
            vflag
          }
        }
        comments {
          votes {
            vflag
          }
        }
        moderating {
          id
          slug
          name
        }
        owner {
          id
          slug
          name
        }
        subreddits {
          id
          slug
          name
        }
        postvotes {
          post {
            title
            id
          }
          id
          vflag
        }
        commentvotes {
          comment {
            id
          }
          id
          vflag
        }
    }

    */