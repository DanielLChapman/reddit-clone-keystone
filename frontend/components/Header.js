import Link from 'next/link';
import Nav from './Nav';


export default function Header() {
  return (
    <>
      <div className="bar">
        <span>
          <Link href="/">App name</Link>
        </span>
        <Nav />
      </div>
    </>
  );
}
