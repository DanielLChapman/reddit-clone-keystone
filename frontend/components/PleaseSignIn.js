import React from 'react';
import { useUser } from './User';
import SignIn from './Signin';

function PleaseSignIn({ children }) {
  const me = useUser();
  if (!me) return <SignIn />;
  return children;
}

export default PleaseSignIn;
