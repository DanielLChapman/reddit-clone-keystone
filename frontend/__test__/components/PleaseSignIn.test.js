import { act, render, screen } from '@testing-library/react';
import {MockedProvider} from '@apollo/react-testing';
import Header from '../../components/Header';
import { signedInMocks, signedInMocksWithPosts } from '../helpers/generalMocks';
import PleaseSignIn from '../../components/PleaseSignIn';

describe('Runs a test if the user is signed in or not', () => {
    it ('It renders a login page if the user is not signed in', async () => {
        const { container, debug } = render(
            <MockedProvider >
              <PleaseSignIn>
                  <span>Hi</span>
              </PleaseSignIn>
            </MockedProvider>
          );

          await screen.findByText(
            'Sign Into Your Account'
          );

        expect(container).toMatchSnapshot();
    });

    it ('It renders a span with text Hi if logged in', async () => {
        const { container, debug } = render(
            <MockedProvider mocks={signedInMocks}>
              <PleaseSignIn>
                  <span>Hi</span>
              </PleaseSignIn>
            </MockedProvider>
          );

          await screen.findByText(
            'Hi'
          );

        expect(container).toMatchSnapshot();
    });
})