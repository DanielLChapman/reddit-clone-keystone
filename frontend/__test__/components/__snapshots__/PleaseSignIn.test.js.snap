// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Runs a test if the user is signed in or not It renders a login page if the user is not signed in 1`] = `
<div>
  <form
    class="Form-sc-1tx35ym-0 etUUQp"
    method="POST"
  >
    <h2>
      Sign Into Your Account
    </h2>
    <fieldset>
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="Your Email Address"
          type="email"
          value=""
        />
      </label>
      <label
        for="password"
      >
        Password
        <input
          autocomplete="password"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Sign In
      </button>
    </fieldset>
  </form>
</div>
`;

exports[`Runs a test if the user is signed in or not It renders a span with text Hi if logged in 1`] = `
<div>
  <span>
    Hi
  </span>
</div>
`;
